content/sips/sip-366.md

```diff
@@ -16,36 +16,277 @@ proposal: >-
+ * account has an outstanding intent to delegate a new amount of
+ */
+function processIntentToDelegateCollateral(
+    uint128 accountId,
+    uint128 poolId,
+    address collateralType
+) public;
+```
+
+The `processIntentToDelegateCollateral` function should, at minimum, satisfy the following requirements:
+
+1. Allow anyone to call it without requiring permissions, enabling the processing of any open intent.
+2. Verify the existence of the specified intent for the given account, pool, and type of collateral.
+3. Ensure the intent's submission timestamp is greater than the current timestamp plus the applicable delay.
+4. Ensure the intent's submission timestamp does not surpass the current timestamp plus both the delay and window period.
+5. Check that processing the intent won't cause the collateralization ratio to fall below the required issuance ratio.
+6. Update the `escrowedCollateralByAccount` to mirror the amount of collateral in process for the designated account and pool.
+7. Delete the intent nonce from `intentNoncesByAccount` for the involved account.
+8. Delete the intent from `intentByNonce` using the specific nonce.
+9. Update the amount of collateral delegated to the pool according to the intent.
+10. Emit a `DelegateCollateralIntentProcessed` event upon successful processing.
+11. Execute steps 2 to 11 for every intent ready for processing for the mentioned account, pool, and collateral type.
+
+> The order by which the intents are processed should not matter so long as the `escrowedCollateralByAccount` mapping field is properly maintained.
+
+##### Edge Cases
+
+If an attempt is made to process an intent but constraints such as issuance ratio restrictions and debt fluctuations prevent the full amount of collateral from being processed, then the maximum permissible amount of collateral should be determined and processed as allowed.
+
+#### Delegating Collateral Entry Point
+
+The `delegateCollateral` function is designed to invoke `requestDelegateCollateral` and/or `processDelegateCollateral` as needed, ensuring compatibility with the pre-existing `delegateCollateral` function. Initially, it aims to declare an intent to delegate a new amount of collateral based on the provided parameters. Subsequently, it seeks to process any existing intents linked to the specified account, pool, and type of collateral.
+
+If an account has a ready-to-process intent but does not wish to declare a new one, it should directly use `processDelegateCollateral`.
+
+Conversely, if an account has no pending intents but wishes to declare a new one, `requestDelegateCollateral` should be utilized directly.
+
+### `LiquidationModule` & `ILiquidationModule`
+
+#### Account Liquidation
+
+In the event an account is liquidated, all outstanding intents to delegate collateral should be canceled/forcibly expired. This can be achieved by batch invalidating all outstanding intent nonces associated with the account within the `liquidate` function.
```

Comment:
Assuming one account has a lot of intents will it be blocking liquidation? Put another way, can we send the mapping to the oubliette instead of deleting individual intents instead?

Reply:
yes, simply updating/clearing the `intentNoncesByAccount` should suffice

Reply:
basically like deleting the intent pointer